inbm/dispatcher-agent/dispatcher/schedule/sqlite_manager.py

```diff
@@ -19,78 +18,254 @@
+            if isinstance(schedule, SingleSchedule):
+                self._create_single_schedule(schedule)
+            elif isinstance(schedule, RepeatedSchedule):
+                self._create_repeated_schedule(schedule)
             else:
-                print("Task is neither a SingleSchedule nor a RepeatedSchedule object.")
+                print("Schedule type is neither a SingleSchedule nor a RepeatedSchedule object.")
         except (sqlite3.OperationalError, sqlite3.InternalError) as e:
-            raise DispatcherException(f"Error connecting to database: {e}")
-        finally:
-            if conn:
-                conn.close()
-
-    def _create_single_schedule_task(self, task: SingleSchedule, conn: sqlite3.Connection) -> None:
-        # Creating the table
-        cur = conn.cursor()
-        # Check if the table already exists, if not, create the table
-        logger.debug("Checking if the table exist...")
-        cur.execute("SELECT name FROM sqlite_master WHERE type='table' AND name='single_task'")
-        if cur.fetchone() is None:
-            logger.debug("Table not exist. Creating the table.")
-            sql = ''' CREATE TABLE single_task (
-                                    request_id text,
-                                    start_time text,
-                                    end_time text,
-                                    manifests text
-                                ); '''
-            conn.execute(sql)  # Execute the SQL command
-
-        # Add the task
-        for manifest in task.manifests:
-            sql = ''' INSERT INTO single_task(request_id, start_time, end_time, manifests)
-                                  VALUES(?,?,?,?) '''
-            logger.debug(f"Adding task: {str(task.request_id), str(task.start_time), str(task.end_time), str(manifest)}")
-            cur.execute(sql, (str(task.request_id), str(task.start_time), str(task.end_time), str(manifest)))
-            conn.commit()
-        logger.debug("Task added.")
+            raise DispatcherException(f"Error connecting to Dispatcher Schedule database: {e}")
+    
+    def _create_single_schedule(self, ss: SingleSchedule) -> None:
+        # Add the schedule to the single_schedule table
+        sql = ''' INSERT INTO single_schedule(request_id, start_time, end_time)
+                                VALUES(?,?,?); '''
+        start_time = None if not ss.start_time else str(ss.start_time)
+        end_time = None if not ss.end_time else str(ss.end_time)
+        try:
+            self._cursor.execute(sql, (ss.request_id, 
+                            start_time, 
+                            end_time))
+        except (sqlite3.IntegrityError, sqlite3.InternalError, sqlite3.OperationalError) as e:
+            raise DispatcherException(f"Error adding single schedule to database: {str(e)}")
+        
+        schedule_id = self._cursor.lastrowid
+        if not schedule_id:
```

Comment:
**<h2>$\color{#FFA500}Improvements: Code Robustness$</h2>**
**Suggestion:** Ensure database transactions are properly managed to maintain data integrity.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
                            try:
                                self._cursor.execute(sql, (ss.request_id, start_time, end_time))
                                schedule_id = self._cursor.lastrowid
                                self._conn.commit()
                            except sqlite3.DatabaseError as e:
                                self._conn.rollback()
                                raise DispatcherException(f"Transaction failed: {e}")
```